packages/web-client/app/utils/wc-provider.ts

```diff
@@ -0,0 +1,526 @@
+}
+
+class WalletConnectProvider extends ProviderEngine {
+  public bridge = 'https://bridge.walletconnect.org';
+  public qrcode = true;
+  public qrcodeModal = QRCodeModal;
+  public qrcodeModalOptions: IQRCodeModalOptions | undefined = undefined;
+  public rpc: IRPCMap | null = null;
+  public rpcWss: IRPCMap;
+  public http: HttpConnection | null = null;
+  public wc: IConnector;
+  public isConnecting = false;
+  public connected = false;
+  public connectCallbacks: any[] = [];
+  public accounts: string[] = [];
+  public chainId!: number;
+  public rpcUrl = '';
+  public websocketProvider!: WebsocketProvider;
+  public networkId!: number;
+  public infuraId?: string;
+
+  constructor(opts: ICardstackWalletConnectProviderOptions) {
+    super({
+      blockTracker: new BlockTracker({
+        provider: (new WebsocketProvider(
+          opts.rpcWss[opts.chainId!]
+        ) as unknown) as Provider,
+      }),
+    });
+    let rpcWss: IRPCMap = opts.rpcWss || null;
+    let chainId: number = opts.chainId!;
+    let websocketProvider = new WebsocketProvider(rpcWss[chainId]);
+    //@ts-ignore allow assignment of new property
+    websocketProvider.sendAsync = function <T, U>(
+      this: WebsocketProvider,
+      req: JsonRpcRequest<T>,
+      cb: (err: Error, res: JsonRpcResponse<U>) => void
+    ): void {
+      this.send(
+        (req as unknown) as JsonRpcPayload,
+        cb as (error: Error | null, result?: unknown) => void
+      );
+    };
+    this.rpcWss = rpcWss;
+    this.chainId = chainId;
+    this.websocketProvider = websocketProvider;
+    this.bridge = opts.connector
+      ? opts.connector.bridge
+      : opts.bridge || 'https://bridge.walletconnect.org';
+    this.qrcode = typeof opts.qrcode === 'undefined' || opts.qrcode !== false;
+    this.qrcodeModal = opts.qrcodeModal || this.qrcodeModal;
+    this.qrcodeModalOptions = opts.qrcodeModalOptions;
+    this.wc =
+      opts.connector ||
+      new WalletConnect({
+        bridge: this.bridge,
+        qrcodeModal: this.qrcode ? this.qrcodeModal : undefined,
+        qrcodeModalOptions: this.qrcodeModalOptions,
+        storageId: opts?.storageId,
+        signingMethods: opts?.signingMethods,
+        clientMeta: opts?.clientMeta,
+      });
+    this.rpc = opts.rpc || null;
+    if (
+      !this.rpc &&
+      (!opts.infuraId ||
+        typeof opts.infuraId !== 'string' ||
+        !opts.infuraId.trim())
+    ) {
+      throw new Error(
+        'Missing one of the required parameters: rpc or infuraId'
+      );
+    }
+    this.infuraId = opts.infuraId || '';
+    this.initialize();
+
+    let socket = new WebSocket(this.rpcWss[this.chainId], []);
+    socket.addEventListener('close', function () {
+      console.log('close', ...arguments);
+    });
+    socket.addEventListener('message', function () {
+      console.log('message', ...arguments);
+    });
+    socket.addEventListener('open', function () {
+      console.log('open', ...arguments);
+    });
```

Comment:
We could probably remove this code, unless we want to keep it for some kind of logging?